src/ext/java/org/opentripplanner/ext/siri/SiriTimetableSnapshotSource.java

```diff
@@ -897,8 +899,25 @@ private boolean addTripToGraphAndBuffer(final String feedId, final Graph graph,
         pattern.getScheduledTimetable().getTripTimes().clear();
 
         // Add to buffer as-is to include it in the 'lastAddedTripPattern'
+        //TODO - Should this update be done twice?
         buffer.update(pattern, updatedTripTimes, serviceDate);
 
+        // Add TripOnServiceDate to buffer if a dated service journey id is supplied in the SIRI message
+        if(estimatedVehicleJourney.getDatedVehicleJourneyRef() != null || estimatedVehicleJourney.getFramedVehicleJourneyRef().getDatedVehicleJourneyRef() != null) {
+            TripAlteration tripAlteration;
+            var dsjId = estimatedVehicleJourney.getDatedVehicleJourneyRef() == null ? estimatedVehicleJourney.getFramedVehicleJourneyRef().getDatedVehicleJourneyRef(): estimatedVehicleJourney.getDatedVehicleJourneyRef().getValue();
+
+            FeedScopedId datedServiceJourneyId = new FeedScopedId(feedId,
+                    dsjId
+            );
+            var tripOnServiceDate = new TripOnServiceDate(datedServiceJourneyId,
+                            trip,
+                            serviceDate,
+                    null);
+
+            buffer.addLastAddedTripOnServiceDate(trip, serviceDate,datedServiceJourneyId, tripOnServiceDate);
+        }
+
```

Comment:
@lassetyr Can you check that this part is ok?

Reply:
The [Nordic SIRI profile](https://enturas.atlassian.net/wiki/spaces/PUBLIC/pages/637370392/SIRI-ET#SIRI-ET-EstimatedVehicleJourney) has not yet landed on how the DSJ-concept should be handled. My answer here is based on the current SIRI Profile, and how it is intended to be used.

In the profile, the field 
`estimatedVehicleJourney.getFramedVehicleJourneyRef().getDatedVehicleJourneyRef()` is defined to contain a ServiceJourneyId, and 
`estimatedVehicleJourney.getFramedVehicleJourneyRef().getDataFrameRef()` the corresponding date.

The field DatedVehicleJourneyRef is not included in the profile, and - in my opinion - it is likely that this field will be defined as the place to use DSJ, but this has to be decided in the Nordic SIRI Profile-process for DSJ.

**However** - both these examples are to be used when the SIRI ET message references an already existing trip. In this case - where the SIRI ET represents an added trip (a trip not present in the planned data) these fields should not be used at all - according to the profile. For an added trip, the "trip-id" (currently ServiceJourneyId) should be in the field `EstimatedVehicleJourneyCode`, and a reference to the replaced trip should be in `VehicleJourneyRef` ( e.g. a railReplacementBus-scenario). The type of id to specify in `VehicleJourneyRef` is not defined - this is probably also something that needs to be decided, but a DSJ is a likely candidate.

Reply:
This breaks the functionality to add trips by causing a NullPointerException.

It is the statement `estimatedVehicleJourney.getFramedVehicleJourneyRef().getDatedVehicleJourneyRef()` that fails.